STJS-586 sentry

wirus15 opened the pull request.

pgs-gmartowski commented on src/client/config/ConfigResolver.ts:
```diff
@@ -50,7 +50,8 @@ export class ConfigResolver {
       translations: this._getValueOrDefault(config.translations, DefaultConfig.translations),
-      visaCheckout: this._setVisaCheckoutConfig(config.visaCheckout, DefaultConfig.visaCheckout)
+      visaCheckout: this._setVisaCheckoutConfig(config.visaCheckout, DefaultConfig.visaCheckout),
+      sentryDsn: this._getValueOrDefault(config.sentryDsn, '')
```
I think this should be in alphabetical order, if we had such convention before

pgs-gmartowski replied:
let's put default value of `sentryDsn` into `DefaultConfig` as well

pgs-gmartowski commented on src/shared/services/sentry/EventScrubber.spec.ts:
```diff
@@ -0,0 +1,42 @@
+import { Event } from '@sentry/types';
+import { EventScrubber } from './EventScrubber';
+
+describe('EventScrubber', () => {
```
I was using `//given // when // then` convention before; should we continue this or leave it ?

pgs-gmartowski commented on src/shared/services/sentry/EventScrubber.spec.ts:
```diff
@@ -0,0 +1,42 @@
+
+    const result = eventScrubber.scrub(event);
+
+    expect(result.extra.config).toEqual({
```
you're using only `config` property here, so I would do sth like:
`
    const { config }= eventScrubber.scrub(event).extra;

    expect(config).toEqual({
`
but it's nth :)

pgs-gmartowski commented on src/shared/services/sentry/EventScrubber.spec.ts:
```diff
@@ -0,0 +1,42 @@
+
+    const result = eventScrubber.scrub(event);
+
+    expect(result.request.url).toBe('https://webservices.securetrading.net?foo=bar&jwt=*****&xyz=abc');
```
looks like `https://webservices.securetrading.net?foo=bar` is repeated twice; maybe it's good to use template strings here

wirus15 commented:
@pgs-gmartowski I made changes from your review. Plus I moved the sentry dsn to environment and added `errorReporting` property to config, like @james-simpson-ST suggested.

pgs-gmartowski commented on example/json/config.json:
```diff
@@ -59,6 +59,7 @@
   "deferInit": false,
   "disableNotification": false,
   "errorCallback": null,
+  "errorReporting": false,
```
I think James suggested that it should be named `debug`

wirus15 replied:
I remember he suggested `debug` but I thought `errorReporting` better describes what this property does. We could use `debug` to put more verbose `console.logs`.

pgs-gmartowski commented on src/shared/services/sentry/SentryService.ts:
```diff
@@ -0,0 +1,35 @@
+      dsn,
+      environment: this.sentryContext.getEnvironmentName(),
+      release: this.sentryContext.getReleaseVersion(),
+      whitelistUrls: ['https://webservices.securetrading.net'],
```
just wondering if such constant values should be somewhere outside of class, or inside as static properties ?
What should be our approach ?

wirus15 replied:
good point 👍
